GameServerLib/GameObjects/AttackableUnits/AI/ObjAIBase.cs

```diff
@@ -691,6 +691,14 @@ public void TeleportTo(float x, float y)
                 x = MovementVector.TargetXToNormalFormat(_game.Map.NavigationGrid, x);
                 y = MovementVector.TargetYToNormalFormat(_game.Map.NavigationGrid, y);
             }
+            
+            // If the unit is a champion, notify all listeners of their movement. 
+            // @TODO: Maybe add a OnChampionTeleport event?
+            if(this is IChampion)
```

Comment:
This is generally considered as a bad practice (https://en.wikipedia.org/wiki/Downcasting). You want Champion to implement TeleportTo and there it will do that and call the base class which is ObjAIBase.